### PR TITLE
MRG: kwarg mask_type like MNE-C

### LINKED ISSUE #3284: correct the API usage of mask in mne.find_events to be consistent with mne-c, and more user-friendly
while I was addressing a comment on the listserv, I noticed that although the masking is correct per its instructions, it is not consistent with the mne-c way of handling the trigger mask. I also take blame because I added the feature. in essence, I inverted the meaning of the ones and zeros, which makes constructing the mask much more difficult.
 this is how it should be:
> --digtrigmask <number>
> Mask to be applied to the trigger channel values before considering
> them. This option is useful if one wants to set some bits in a don’t
> care state. For example, some finger response pads keep the trigger
> lines high if not in use, i.e., a finger is not in place. Yet, it is convenient
> to keep these devices permanently connected to the acquisition
> system. The number can be given in decimal or hexadecimal format
> (beginning with 0x or 0X). For example, the value 255 (0xFF)
> means that only the lowest order byte (usually trigger lines 1 - 8 or
> bits 0 - 7) will be considered.

I propose that we make it our mask option consistent with this. it makes its use case more user-friendly, and fixes the inconsistency issue. 

a real example, a user's triggers ranged from 0-32 and are only described on the first six bits of the trigger channels. in order to isolate these values, we need to mask the upper bits. the current implementation requires `mask=2**17 -32` whereas the mne-c way, and the way I propose would have it be `mask=32`.

What are your thoughts?
cc: @agramfort, @Eric89GXL

### PR DESCRIPTION
**Pull request by Eric89GXL**
Closes #3284

### CODE REVIEW

**Comment by Eric89GXL** [comment]
CircleCI error seems unrelated, ready for review/merge from my end

File: mne/event.py [hunk 1]
```diff
@@ -209,7 +210,12 @@ def read_events(filename, include=None, exclude=None, mask=0):
         the exclude parameter is ignored.
     mask : int | None
         The value of the digital mask to apply to the stim channel values.
-        The default value is 0. ``None`` skips masking.
+        The default value is None. If None, trigger masking is skipped.
+    mask_type: 'and' | 'not_and'
+        The type of operation between the mask and the trigger.
+        Choose 'and' for MNE-C masking behavior.
```
Comment (mmagnuski) [inline mne/event.py]: I don't find the explanation of `mask` and `mask_type` arguments here very clear. They are explained clearly in docstring of `find_events` - maybe it should be linked/referenced here somehow?

Comment (mmagnuski) [inline mne/event.py]: I mean - after reading the description in `find_events` the description of aruments here is very clear, but not before. :)

File: mne/event.py [hunk 2]
```diff
@@ -633,17 +655,28 @@ def find_events(raw, stim_channel=None, output='onset',
...
 
 
-def _mask_trigs(events, mask):
+def _mask_trigs(events, mask, mask_type):
     """Helper function for masking digital trigger values"""
-    if not isinstance(mask, int):
-        raise TypeError('You provided a(n) %s. Mask must be an int.'
-                        % type(mask))
+    if mask is not None:
+        if not isinstance(mask, int):
```
Comment (mmagnuski) [inline mne/event.py]: slightly offtopic - I think ability to pass `mask` as a boolean array would be quite intuitive (more intuitive maybe).

**Comment by mmagnuski** [comment]
Looks good - so my nitpicking was very limited :)

File: mne/event.py (hunk 1 above)
Comment (Eric89GXL) [inline mne/event.py]: Most of the time people will probably learn about `mask` in `find_events` anyway, I think. I'll improve the links to that doc so we don't duplicate

File: mne/event.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/event.py]: Good for another PR if need be, but I'd assume YAGNI for now

**Comment by Eric89GXL** [comment]
Comments addressed thanks for the look @mmagnuski

**Comment by teonbrooks** [comment]
LGTM +1

**Comment by mmagnuski** [comment]
LGTM too

**Comment by Eric89GXL** [comment]
Thanks for the reviews. Since this already had a +1 or two from the original PR, I'll go ahead and merge.

**Comment by mmagnuski** [comment]
Just in case nobody noticed the nice & round PR number: :tada: for #3500 :)

**Pull request merged**